docs/application/dotnet/guides/nui/styling-controls-with-xaml.md

```diff
@@ -0,0 +1,190 @@
+  PointSize = 20
+};
+
+TextFieldStyle textStyle = new TextFieldStyle()
+{
+  EnableCursorBlink = False
+};
+```
+
+## Define theme using XAML
+
+It is better to use XAML theme files in advanced applications to define UI style for NUI widgets. With XAML files, it is easier to share styles between different applications. From the application architecture point of view, XAML allows for better separation of view definitions from application logic.
+
+### Overview XAML example
+
+ The basic text styles defined in the earlier section can be implemented using XAML as follows:
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<Theme
+  xmlns="http://tizen.org/Tizen.NUI/2018/XAML"
+  xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml">
+
+  <TextLabelStyle x:Key="redTextStyle" TextColor="Red" PointSize="20" />
+</Theme>
+```
+
+### Handle component states
+
+Defining different states of a component is based on creating an `Selector` object. In `Selector` constructor individual states should be specified. 
+
+```csharp
+Selector selector = new Selector<Color>()
+{
+  Normal = Color.Black,
+  Pressed = Color.Blue,
+  Disabled = Color.Red
+};
+```
+
+The equivalent XAML code is as follows:
+
+```xml
+<Selector x:TypeArguments="Color" Normal="Black" Pressed="Blue" Disabled="Red"/>
+```
+
+The selectors work when the control state is enabled in a `View`. By default, the control state is enabled for all controls in `Tizen.NUI.Components` and not for the controls in `Tizen.NUI.BaseComponents`, such as `View` and `TextLabel`. If you want to enable a control state for them, set the `EnableControlState` property to `true`:
+
+```xml
+<ViewStyle x:Key="viewColoredByTouch" EnableControlState="true">
+  <ViewStyle.BackgroundColor>
+      <Selector x:TypeArguments="Color" Normal="Black" Pressed="Blue" Disabled="Red"/>
+  </ViewStyle.BackgroundColor>
+</ViewStyle>
+```
+}
+
+### Load theme in NUI application
+
+To use the theme implemented in XAML you need to do the following three steps: 
+1. Create theme object: `Theme theme = new Theme("PATH")`, where PATH is a full path to XAML file in a project.
+2. Use `ThemeManager` to apply theme: `ThemeManager.apply(theme)`
+3. Use style defined in the XAML file in NUI component by set StyleName parameter: `component.StyleName = "xaml_style_name"
+
+Following example explains above steps for TextLabel NUI component:
+
+```csharp
+using Tizen.NUI; 
+
+namespace NUI_Theme
+{
+  class Program : NUIApplication
+  {
+    protected override void OnCreate()
+    {
+      // Load theme from the xaml file
+      Theme theme = new Theme("full_path_of_the_xaml_file.xaml");
+
+      // Apply it to the current NUI Application
+      ThemeManager.ApplyTheme(theme);
+
+      // Set style name in NUI widget
+      TextLabel textLabel = new TextLabel()
+      textLabel.StyleName = "redTextStyle";
+    }
+  }
+```
+
+### Handle theme changes in runtime
+
+The following methods can be used to handle theme changes in runtime:
```

Comment:
The following methods can be used to handle theme changes in runtime:

- `ThemeManager.ThemeChanged`
- `View.OnThemeChanged`